lib/std/io/tty.zig

```diff
@@ -133,3 +132,39 @@ pub const Config = union(enum) {
         };
     }
 };
+
+/// Obtains the size of a terminal designated by the file descriptor, or null if that wasn't possible,
+/// either because the file descriptor doesn't refer to a terminal or something else.
+pub fn getSize(file_descriptor: File) ?struct { rows: u16, columns: u16 } {
+    if (native_os == .windows) {
+        var info: windows.CONSOLE_SCREEN_BUFFER_INFO = undefined;
+        if (windows.kernel32.GetConsoleScreenBufferInfo(file_descriptor.handle, &info) != windows.FALSE) {
+            // In the old Windows console, info.dwSize.Y is the line count of the
+            // entire scrollback buffer, so we use this instead so that we
+            // always get the size of the screen.
+            const screen_height = info.srWindow.Bottom - info.srWindow.Top;
+            return .{
+                .rows = @intCast(screen_height),
+                .columns = @intCast(info.dwSize.X),
+            };
+        } else {
+            return null;
+        }
+    } else {
+        var winsize: posix.winsize = .{
```

Comment:
In case someone knows: can't this just be initialized to `undefined`? Or are there cases where a field would not be written to by `ioctl`? If only in case of failure that wouldn't matter obviously.